src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+
+        public override bool IsStatic => true;
+
+        public override TypeKind TypeKind => TypeKind.Class;
+
+        private readonly int TypeParametersCount;
+        public override int Arity => TypeParametersCount;
+
+        public readonly SynthesizedFieldSymbol DelegateField;
+
+        private string _name = null;
+        public override string Name => _name;
+
+        private int _index = -1;
+        public int Index => _index;
+
+        public string SortKey { get; }
+
+        private readonly ImmutableArray<TypeParameterSymbol> BuiltTypeParameters;
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => BuiltTypeParameters;
+
+        private MethodGroupConversionCacheFrame(
+                MethodGroupConversionCacheFrameManager manager,
+                MethodSymbol targetMethod,
+                int arity)
+        {
+            Manager = manager;
+            SortKey = SymbolDisplay.ToDisplayString(targetMethod, SymbolDisplayFormat.FullyQualifiedFormat);
+
+            TypeParametersCount = arity;
+
+            var typeParamsBuilder = ArrayBuilder<TypeParameterSymbol>.GetInstance();
+            for ( int i = 0; i < arity; i++ )
+            {
+                typeParamsBuilder.Add(new MethodGroupConversionCacheFrameTypeParameter(this, i));
+            }
+            BuiltTypeParameters = typeParamsBuilder.ToImmutableAndFree();
+
+            var fieldType = TypeParameters[TypeParameters.Length - 1];
+            var fieldName = GeneratedNames.MakeMethodGroupConversionCacheDelegateFieldName(targetMethod.Name);
+            DelegateField = new SynthesizedFieldSymbol(this, fieldType, fieldName, isPublic: true, isStatic: true);
+        }
+
+        /// <summary>
+        /// Assign the name and index to be serialized in the module.
+        /// </summary>
+        /// <remarks>This method is only intended to be called from <see cref="MethodGroupConversionCacheFrameManager"/></remarks>
+        internal void AssignNameAndIndex(string name, int index)
+        {
+            _name = name;
+            _index = index;
+        }
+
+        /// <remarks>This method is only intended to be called from <see cref="MethodGroupConversionCacheFrameManager"/></remarks>
+        internal static MethodGroupConversionCacheFrame Create(MethodGroupConversionCacheFrameManager manager, MethodSymbol targetMethod)
+        {
+            Debug.Assert(manager != null);
+            Debug.Assert(targetMethod.IsDefinition);
+
+            return new MethodGroupConversionCacheFrame(manager, targetMethod, CalculateNeededArity(targetMethod));
+        }
+
+        private static int CalculateNeededArity(MethodSymbol targetMethod)
+        {
+            var arity = 1; // the type of the delegate
+
+            arity += targetMethod.Arity;
+
+            for (var containingType = targetMethod.ContainingType; containingType!= null; containingType = containingType.ContainingType)
```

Comment:
Use cast to ```object``` before comparing symbols to null, this avoids an invocation of an overloaded comparison operator.